### ISSUE TITLE
The number of crashes on the release page and the discover page differ significantly

### ISSUE THREAD

**Issue by PayneFuRC** [getsentry/sentry#70570]
### Environment

SaaS (https://sentry.io/)

### Steps to Reproduce

1. There are 9.1k crashes in release page(https://jupiter-ct.sentry.io/releases/?project=1430058&query=24.2.17&statsPeriod=90d)
![image](https://github.com/getsentry/sentry/assets/150770050/4ede3b90-5b6a-47dd-b7cd-df6644cdff7b)

2. There are 3.8k crashes in discover page(https://jupiter-ct.sentry.io/discover/homepage/?field=title&field=event.type&field=project&field=user.display&field=timestamp&name=All+Events&project=1430058&query=release%3Adesktop%4024.2.17%2BRCAppDesktop+error.unhandled%3Atrue&sort=-timestamp&statsPeriod=90d&yAxis=count%28%29)
![image](https://github.com/getsentry/sentry/assets/150770050/fb0740ee-c099-4337-8efa-8124189a1c2d)

3. There's no dropped crashes
4. There's no filter rules

### Expected Result

These two data sets should be the same.

### Actual Result

These two data sets differ significantly.

### Product Area

Releases

**Comment by PayneFuRC** [comment]
And the data from other versions is the opposite, the number of crashes on the release page is lower. If the logic of beforeSend filtered it out, this situation should not exist.
![image (1)](https://github.com/getsentry/sentry/assets/150770050/e670d027-ff3d-418f-87ae-b160d6be8045)
![image (2)](https://github.com/getsentry/sentry/assets/150770050/54e7bc52-3f25-48b7-8129-e5341e97b5a5)

**Comment by nhsiehgit** [comment]
Hi @PayneFuRC 

Thanks for the report!
It looks like our releases page may be deriving this count from somewhere unexpectedly different than our discover page.

I might err on trusting the discover query for the crash rate for now while we dig to understand where this discrepancy is stemming from.

will create an internal ticket for us to track

**Comment by nhsiehgit** [comment]
Hi @PayneFuRC 

Dug into this a bit more
So it looks like the releases page _is_ filtering these events for `'session.status' === 'crashed'` as opposed to simply `error.unhandled:true`

So the releases count actually _will_ be the more proper count!
I'll work with the team to see how we can surface this in a more clear manner.

**Comment by PayneFuRC** [comment]
@nhsiehgit How can I filter for such events? It seems like there's no search condition(`'session.status' === 'crashed'`) for this. Will you support this kind of search condition in the future?

**Comment by nhsiehgit** [comment]
@PayneFuRC 

I'm asking around now.
It seems like discover only queries for Errors/Transaction events, whereas `session`s may simply be metrics with no underlying events

that being said, aside from the 2 pages querying different datasets, it sounds like there may also be a bug in how the electron SDK is reporting sessions 🤔 
Apologies for the non-answer here. I'll try to update this issue as we get more information

**Comment by PayneFuRC** [comment]
@nhsiehgit I have also submitted a ticket to Sentry/Electron regarding the session crash report. Please help track this issue as well.

https://github.com/getsentry/sentry-electron/issues/941

**Comment by masinette** [comment]
Related to Intercom ticket #98396866

**Comment by bcoe** [comment]
The crash rate information shown on the releases page is based on the [release health API](https://docs.sentry.io/product/releases/health/), a standalone dataset that tracks telemetry about healthy sessions, crashed sessions, etc., whereas the results shown on the discover page are based on accepted errors.

While these #s should usually be close, there are reasons why they might drift apart. The number of accepted errors can vary based on:

* [Filtering rules](https://docs.sentry.io/platforms/javascript/configuration/filtering/).
* Malformed errors, e.g, payloads that are too large.
* Quota.

tldr; the way that Discover and Releases are implemented today, it's expected behavior that there can be drift between the counts in the two datasets.

A good starting point for debugging is looking at [stats](https://docs.sentry.io/product/stats/) are there error events being dropped throughout a month, this would correlate to a drift in the two values.

----

I realize this behaviour can be confusing an unintuitive. How to modernize the session health / releases dataset is an active discussion we've been having internally at Sentry (_making the drift between session status and error count will be part of this conversation!_).
